lightning/src/onion_message/messenger.rs

```diff
@@ -516,18 +583,23 @@ where
 	pub fn send_onion_message<T: OnionMessageContents>(
```

Comment:
Wondering if this should still be `pub`? We send through the handlers, so this is not used internally outside of this module. In this PR, we need to examine the return value to decide if a connection is needed. So users calling this won't trigger a connection request nor will the message be buffered for themexcept if we previously decided that a connection is needed.

Reply:
The interface here isn't congruent with other sending, FWIW. This takes an `OnionMessagePath` instead of a `Destination`, so the assumption is that the `MessageRouter` has already been called prior to calling `send_onion_message`. We'd probably want to do message routing internally, I'd imagine?